content/doc/book/pipeline/jenkinsfile.adoc

```
@@ -420,10 +428,10 @@ apply to all steps within the Pipeline.
 given environment variables to steps within the `stage`.
 
 
-==== Setting environment variables dynamically 
+==== Setting environment variables dynamically
 
-In the case where environment variable need to be set dynamically at run time this can be done with the use of a shell scripts (`sh`), Windows Batch Script (`bat`) or PowerShell Script (`powershell`). 
-Each script can either `returnStatus` or `returnStdout`.  
+Environment variables can be set at run time with shell scripts (`sh`), Windows batch scripts (`bat`) and PowerShell scripts (`powershell`).
```

Environment variables in this context are set by the Pipeline and *used by* the scripts.  As far as I know, there isn't an obvious way for a script to set an environment variable and have it persist in the Pipeline for use in later scripts.
```suggestion
Environment variables can be set at run time and can be used by shell scripts (`sh`), Windows batch scripts (`bat`) and PowerShell scripts (`powershell`).
```